src/couch_jobs/test/couch_jobs_tests.erl

```diff
@@ -0,0 +1,607 @@
+            couch_jobs:update(Tx, Job)
+        end)),
+
+        ?assertMatch(#{data := #{<<"x">> := 1}, state := running},
+            get_job(T, J)),
+
+        ?assertMatch({ok, #{job := true}}, fabric2_fdb:transactional(fun(Tx) ->
+            couch_jobs:update(Tx, Job, #{<<"x">> => 2})
+        end)),
+
+        % check json validation for bad data in update
+        ?assertMatch({error, {json_encoding_error, _}},
+            fabric2_fdb:transactional(fun(Tx) ->
+                couch_jobs:update(Tx, Job, #{1 => 1})
+            end)),
+
+        ?assertMatch(#{data := #{<<"x">> := 2}, state := running},
+            get_job(T, J)),
+
+        % Finish may update the data as well
+        ?assertEqual(ok, couch_jobs:finish(?TX, Job, #{<<"x">> => 3})),
+        ?assertMatch(#{data := #{<<"x">> := 3}, state := finished},
+            get_job(T, J))
+    end).
+
+
+resubmit_enqueues_job(#{t1 := T, j1 := J}) ->
+    ?_test(begin
+        ok = couch_jobs:add(?TX, T, J, #{}),
+        {ok, Job1, #{}} = couch_jobs:accept(T),
+        ?assertMatch({ok, _}, couch_jobs:resubmit(?TX, Job1, 6)),
+        ?assertEqual(ok, couch_jobs:finish(?TX, Job1)),
+        ?assertMatch(#{state := pending, stime := 6}, get_job(T, J)),
+        {ok, Job2, #{}} = couch_jobs:accept(T),
+        ?assertEqual(ok, couch_jobs:finish(?TX, Job2)),
+        ?assertMatch(#{state := finished}, get_job(T, J))
+    end).
+
+
+add_with_scheduled_time(#{t1 := T, j1 := J1, j2 := J2}) ->
+    ?_test(begin
+        ?assertEqual(ok, couch_jobs:add(?TX, T, J1, #{}, 5)),
+        ?assertEqual(ok, couch_jobs:add(?TX, T, J2, #{}, 3)),
+        ?assertMatch({ok, #{id := J}, _} when J =:= J1 orelse J =:= J2,
+            couch_jobs:accept(T, #{max_sched_time => 3})),
+        ?assertMatch({ok, #{id := J}, _} when J =:= J1 orelse J =:= J2,
+            couch_jobs:accept(T, #{max_sched_time => 9}))
+    end).
+
+
+resubmit_custom_schedtime(#{t1 := T, j1 := J}) ->
+    ?_test(begin
+        ?assertEqual(ok, couch_jobs:add(?TX, T, J, #{}, 7)),
+        {ok, Job, #{}} = couch_jobs:accept(T),
+        ?assertMatch({ok, _}, couch_jobs:resubmit(?TX, Job, 9)),
+        ?assertEqual(ok, couch_jobs:finish(?TX, Job)),
+        ?assertMatch(#{stime := 9, state := pending}, get_job(T, J))
+    end).
+
+
+accept_max_schedtime(#{t1 := T, j1 := J1, j2 := J2}) ->
+    ?_test(begin
+        ok = couch_jobs:add(?TX, T, J1, #{}, 5),
+        ok = couch_jobs:add(?TX, T, J2, #{}, 3),
+        ?assertMatch({ok, #{id := J2}, _}, couch_jobs:accept(T,
+            #{max_sched_time => 3})),
+        ?assertMatch({ok, #{id := J1}, _}, couch_jobs:accept(T,
+            #{max_sched_time => 9}))
+    end).
+
+
+accept_random_ids(#{t1 := T}) ->
+    ?_test(begin
+        JobCount = 25,
+        Jobs = [fabric2_util:uuid() || _ <- lists:seq(1, JobCount)],
+        [couch_jobs:add(?TX, T, J, #{}) || J <- Jobs],
+        AcceptOpts = #{max_sched_time => 0, random_ids => true},
```

Comment:
This random_ids thing seems to be to grab any random job? Seems like not the best name. Earlier reading I thought it was something about "assign a random id". It also appears that max_sched_time doesn't work in conjunction with random_ids since its eventually hard coded as 0 in `couch_jobs_pending:dequeue/4`. We should either error out somewhere in there or support the combination though I can't say that the combination is making sense off the top of my head.

Reply:
Also the comment in couch_jobs_pending now makes more sense, we'll probably want to re-think how we expose that API since we're basically requiring the UUID but the guidance on that is buried a few layers deeper than the average API user is gonna be reading.

Reply:
Yeah it uniformly picks a random ids from the list of ids. I'll add an API error so that accept fails if random_ids is used and max_sched_time is not 0

Reply:
Also make sure you allow for max_sched_time to be unspecified.